### PR TITLE
adding shellcheck action in cliworkflow

### PR DESCRIPTION
**Pull request by ulshell**
Committing issue #549

### CODE REVIEW

**Comment by ulshell** [comment]
Kindly review @ivotron

File: ci/cli.workflow [hunk 1]
```diff
@@ -63,3 +63,8 @@ action "end" {
     "test dry-run"
   ]
 }
+
+action "Shellcheck" {
+  uses = "ci/test/shellcheck@master"
```
Comment (ivotron) [inline ci/cli.workflow]: this has to point to the actions repo, i.e.:

```hcl
uses = "actions/bin/shellcheck@master"
```

**Comment by ulshell** [comment]
Hello @ivotron,
Thanks for reviewing I have made some changes. Kindly see to it.
Thank you,
Utkarsh Mishra

File: ci/cli.workflow [hunk 2]
```diff
@@ -63,3 +63,8 @@ action "end" {
     "test dry-run"
   ]
 }
+
+action "Shellcheck" {
+  uses = "actions/bin/shellcheck@master"
+  args = "*.sh"
```
Comment (ivotron) [inline ci/cli.workflow]: this needs to reference the bash files in the test folder, i.e. `./ci/test/*`

**Comment by ulshell** [comment]
Now I think it will be correct but travis fails kindly suggest anything to overcome.

**Comment by ivotron** [comment]
hey @ulshell , looks like we get shellcheck errors that need to be fixed, reported [here](https://travis-ci.org/systemslab/popper/jobs/510848616#L869-L1137). Do you feel comfortable to work on fixing those? You can test locally by running `popper run Shellcheck` on your machine.

**Comment by ulshell** [comment]
Hello @ivotron  I need some help in this issue. I want some guidance to approach this issue.

**Comment by arshul** [comment]
@ulshell you need to modify the the scripts in `./ci/test/` directory to pass the shell-check. You can run the run the Shellcheck action on your local by `popper run 'Shellcheck'` to see what is causing the error. 
To start off with, you can find the solution for `SC1091: Not following: ./common was not specified as input (see shellcheck -x)` error [here](https://github.com/koalaman/shellcheck/wiki/SC1091). Similarly you can find the solutions for every error you see by using the SC code e.g. for `SC2086: Double quote to prevent globbing and word splitting` you can go to [https://github.com/koalaman/shellcheck/wiki/SC2086](https://github.com/koalaman/shellcheck/wiki/SC2086). I hope this helps, do let me know if you have any concerns.

**Comment by ivotron** [comment]
@ulshell please take a look at the comments by @arshul . Let us know if something is not clear.

**Comment by ulshell** [comment]
Thanks @arshul for helping me out

**Comment by ivotron** [comment]
closing in favor of #572

**Pull request closed without merging**